pkg/descheduler/evictions/evictions_test.go

```diff
@@ -33,42 +38,69 @@ import (
 	"sigs.k8s.io/descheduler/test"
 )
 
+const (
+	notFoundText    = "pod not found when evicting \"%s\": pods \"%s\" not found"
+	tooManyRequests = "error when evicting pod (ignoring) \"%s\": Too many requests: too many requests"
+)
+
 func TestEvictPod(t *testing.T) {
 	ctx := context.Background()
 	node1 := test.BuildTestNode("node1", 1000, 2000, 9, nil)
 	pod1 := test.BuildTestPod("p1", 400, 0, "node1", nil)
 	tests := []struct {
 		description string
 		node        *v1.Node
-		pod         *v1.Pod
-		pods        []v1.Pod
-		want        error
+		evictedPod  *v1.Pod
+		pods        []runtime.Object
+		wantErr     bool
+		wantErrMsg  string
 	}{
 		{
 			description: "test pod eviction - pod present",
 			node:        node1,
-			pod:         pod1,
-			pods:        []v1.Pod{*pod1},
-			want:        nil,
+			evictedPod:  pod1,
+			pods:        []runtime.Object{pod1},
+			wantErr:     false,
+		},
+		{
+			description: "test pod eviction - pod absent (not found error)",
+			node:        node1,
+			evictedPod:  pod1,
+			pods:        []runtime.Object{test.BuildTestPod("p2", 400, 0, "node1", nil), test.BuildTestPod("p3", 450, 0, "node1", nil)},
+			wantErr:     true,
+			wantErrMsg:  fmt.Sprintf("pod not found when evicting \"%s\": pods \"%s\" not found", pod1.Name, pod1.Name),
 		},
 		{
-			description: "test pod eviction - pod absent",
+			description: "test pod eviction - pod absent (too many requests error)",
 			node:        node1,
-			pod:         pod1,
-			pods:        []v1.Pod{*test.BuildTestPod("p2", 400, 0, "node1", nil), *test.BuildTestPod("p3", 450, 0, "node1", nil)},
-			want:        nil,
+			evictedPod:  pod1,
+			pods:        []runtime.Object{test.BuildTestPod("p2", 400, 0, "node1", nil), test.BuildTestPod("p3", 450, 0, "node1", nil)},
+			wantErr:     true,
+			wantErrMsg:  fmt.Sprintf("error when evicting pod (ignoring) \"%s\": Too many requests: too many requests", pod1.Name),
 		},
 	}
 
 	for _, test := range tests {
-		fakeClient := &fake.Clientset{}
-		fakeClient.Fake.AddReactor("list", "pods", func(action core.Action) (bool, runtime.Object, error) {
-			return true, &v1.PodList{Items: test.pods}, nil
+		t.Run(test.description, func(t *testing.T) {
+			fakeClient := fake.NewClientset(test.pods...)
+			fakeClient.PrependReactor("create", "pods/eviction", func(action core.Action) (handled bool, ret runtime.Object, err error) {
+				if test.wantErrMsg == fmt.Sprintf(notFoundText, test.evictedPod.Name, test.evictedPod.Name) {
```

Comment:
What about to pass `k8serrors.XXX` directly? I.e.
```
[]struct {
	description string
	node        *v1.Node
	evictedPod  *v1.Pod
	pods        []runtime.Object
	wantErr     error
}{
	...
	{
		description: "test pod eviction - pod absent (not found error)",
		node:        node1,
		evictedPod:  pod1,
		pods:        []runtime.Object{test.BuildTestPod("p2", 400, 0, "node1", nil), test.BuildTestPod("p3", 450, 0, "node1", nil)},
		wantErr:  k8serrors.NewNotFound(v1.Resource("pods"), pod1.Name),
	},
}
...
fakeClient.PrependReactor("create", "pods/eviction", func(action core.Action) (handled bool, ret runtime.Object, err error) {
	return true, nil, test.wantErr
})
```
The original wantErr bool can be turned into `wantErr != nil`.

Reply:
Because we return using the `fmt.Error`, we cannot directly use `k8serrors.XXX`, but we can simplify it to directly use `fmt.Error` to return

https://github.com/kubernetes-sigs/descheduler/blob/da52983b2773cfb5d014a6d3f7df0d0a0b3bf6bf/pkg/descheduler/evictions/evictions.go#L213-L221